### DISCUSSION TITLE
Rotation issue with IPF Maps

### DISCUSSION THREAD

**Discussion by SunnySharma200** [mtex-toolbox/mtex#1592] [Ask Anything]
Hi @ralfHielscher,

I am Sunny Sharma, a Ph.D. student from IIT Madras. Working in additive manufacturing of Inconel 718 alloy.

I am new to this MTEX software and face issues related to IPF Maps.

I have 2 types of samples: as-deposited and heat treated. When I run the MTEX code to plot the IPF map for the as-deposited sample matches with the optical micrograph, but when it comes to the solution treated, it is not matching. It looks like the IPF map is rotated 90 deg.

I have attached the figure and the code for your reference; kindly help me with this.
![IPFquery](https://user-images.githubusercontent.com/70128690/195489483-2c854b87-dde9-4d5a-ab9d-e8e6b9469aeb.png)
```
### CODE:
**%% Import Script for EBSD Data**
%
% This script was automatically created by the import wizard. You should
% run the whole script or parts of it to import your data. There
% is no problem in making any changes to this script.

**%% Specify Crystal and Specimen Symmetries**
clc
clear 
close all

% crystal symmetry
CS = {... 
  'notIndexed',...
  crystalSymmetry('432', [3.6 3.6 3.6], 'mineral', 'Nickel', 'color', [0.53 0.81 0.98])};
SS = specimenSymmetry('1'); % triclinic symmetry
% plotting convention
setMTEXpref('xAxisDirection','north');
setMTEXpref('zAxisDirection','outOfPlane');

%% Specify File Names

% path to files
pname = '/home/iitchennai/Dropbox/AMoM/LAM_EBSD Journal Work/E587';

% which files to be imported
fname = [pname '/Sample 1 front plane.ang'];

%% Import the Data

% create an EBSD variable containing the data
ebsd = EBSD.load(fname,CS,'interface','ang',...
  'convertEuler2SpatialReferenceFrame','setting 2');
%% Phase Map
figure(1)
plot(ebsd);
%% Plotting of IPF map
%define orientation color key
ipfKey = ipfTSLKey(ebsd.CS);
figure(2)
plot(ipfKey);
color = ipfKey.orientation2color(ebsd.orientations);
figure(3)
plot(ebsd,color,'coordinates','off','MicronBar','on','FontSize',20);
f=gcm;
mp=getappdata(f.currentAxes,'mapPlot');

mp.micronBar % properties which can be set
mp.micronBar.backgroundAlpha=1; % change transparency
mp.micronBar.backgroundColor= "black" ; % change background color
mp.micronBar.lineColor = "white"; % change bar color

mp.ax.FontSize = 20; % set legend to old fontsize
setMTEXpref('FontSize',20); % restore old fontsize
mP = getappdata(gca,'mapPlot');

mP.micronBar.length = 50;
mP.micronBar.update;
```

**Comment by zmichels** [comment]
Maybe you can include/link/upload your data (or some fragment)? It might be easier for folks to help troubleshoot.

**Reply by SunnySharma200** [reply]
Hello @zmichels 

Can I share the .ang files as a google drive link?

I hope that will help you in resolving my doubt.

Thanks
@SunnySharma200

**Comment by kilir** [comment]
Hi,
I have a hard time finding any feature from the optical image in your EBSD mal. Are you sure that they are the same area?
Also, I assume you x,z on your optical image are rather arbitrary directions, unrelated to your specimen coordinates of your ebsd data, correct?
Cheers,
Rüdiger

**Reply by SunnySharma200** [reply]
Hello, thanks for responding,

The area in OM and EBSD are not the same, but from the same surface; hence the morphology will be almost similar.
 
Yes, here, Z is the built direction (BD), and X can be called the transverse direction (TD) 

Thanks 
Sunny

**Comment by AmitS07** [comment]
Hi Sunny, 
I have also faced the same issue. It was issue with the specimen reference frame. If you have done EBSD in TSL, then you have to choose the "use OSC interface flag 'convertSpatial2EulerReferenceFrame" and the MTEX plotting Convention should be 4th one as indicted in the below figure (4th from left side), which means Z into the plane and Y-East and X-south. 
![image](https://user-images.githubusercontent.com/67289970/197033262-041fb859-63a6-4be6-a6d3-bcc280a2fff1.png)

Here is the another image showing the convention to plot EBSD data from different commercial software, like TSL/HKL to match the same data in MTEX.
![Picture1](https://user-images.githubusercontent.com/67289970/197032027-3a9e3ed9-d196-4ca0-8a29-8682e1d44f6a.png)

Hope this will help.
